libcudacxx/docs/extended_api/synchronization_primitives/barrier.md

```diff
@@ -60,6 +61,43 @@ Programs shall ensure that this transformation would not introduce errors, for
 Under CUDA Compute Capability 6 (Pascal) or prior, an object of type
   `cuda::barrier` or `cuda::std::barrier` may not be used.
 
+## Shared memory barriers with transaction count
+
+Starting with the Hopper architecture (CUDA Compute Capability 9), a
+`cuda::barrier` object located in shared memory supports a new count, called
+[tx-count](https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#tracking-asynchronous-operations-by-the-mbarrier-object),
+which is used for tracking the completion of asynchronous memory operations or
+transactions. The tx-count tracks the number of asynchronous transactions, in
+units specified by the asynchronous memory operation (typically bytes), that are
+outstanding and yet to be complete.
+
+The tx-count of `cuda::barrier` must be set to the total amount of asynchronous
+memory operations, in units as specified by the asynchronous operations, to be
+tracked by the current phase. This can be achieved with the
+`cuda::device::barrier_arrive_tx` function call. Upon completion of each of the
+asynchronous operations, the tx-count of the `cuda::barrier` will be updated and
+thus progress the `cuda::barrier` towards the completion of the current phase.
+This may complete the current phase.
```

Comment:
This needs to cover that `barrier_arrive_tx` does not need to be called _before_ any async op, but rather that both can happen concurrently, etc.

Reply:
This is now mentioned in the barrier phase completion section, i.e., that they operate atomically.